internal/supervisor/cluster_installation_test.go

```diff
@@ -23,23 +22,51 @@ type mockClusterInstallationStore struct {
 	Installation                            *model.Installation
 	ClusterInstallation                     *model.ClusterInstallation
 	UnlockedClusterInstallationsPendingWork []*model.ClusterInstallation
+	Clusters                                []*model.Cluster
+	Installations                           []*model.Installation
 	ClusterInstallations                    []*model.ClusterInstallation
 
 	UnlockChan                     chan interface{}
 	UpdateClusterInstallationCalls int
 }
 
 func (s *mockClusterInstallationStore) GetCluster(clusterID string) (*model.Cluster, error) {
-	return s.Cluster, nil
+	if s.Cluster != nil {
```

Comment:
I know that this is the mock, but isn't necessary here to check that `s.Cluster.ID == clusterID`? Just wondering.

Reply:
Previously, this method returned the cluster regardless of `clusterID`.

```go
func (s *mockClusterInstallationStore) GetCluster(clusterID string) (*model.Cluster, error) {
	return s.Cluster, nil
}
```

`S.Cluster` is set if the test contains a single cluster. So `clusterID` does not require verification. 

However, for this new test, the list of clusters is set in `s.Clusters` ignoring `s.Cluster`

Reply:
Gotcha 👍